DoctorWho.Web/Controllers/DoctorsController.cs

```diff
@@ -0,0 +1,64 @@
+﻿using DoctorWho.Db.Repositories;
+using DoctorWho.Web.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DoctorWho.Web.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+
+    public class DoctorsController : ControllerBase
+    {
+        private readonly DoctorRepository _doctorRepository;
+        public DoctorsController(DoctorRepository doctorRepository)
+        {
+            _doctorRepository = doctorRepository;
+        }
+
+        [HttpGet]
+        public ActionResult<IEnumerable<DoctorDto>> GetAllDoctors()
+        {
+            var doctorDtos = _doctorRepository.GetAllDoctors();
+
+            return Ok(doctorDtos);
+        }
+
+        [HttpPost("upsert")]
+        public ActionResult<DoctorDto> UpsertDoctor([FromBody] DoctorDto doctorDto)
+        {
+
+            if (doctorDto.DoctorId > 0)
+            {
+                var doctorEntity = _doctorRepository.GetDoctorById(doctorDto.DoctorId);
+
+                if (doctorEntity == null)
+                {
+                    return NotFound("Doctor not found");
+                }
+
+                _doctorRepository.UpdateDoctor(doctorEntity, doctorDto);
+                return NoContent();
+            }
+
+            var createdDoctorDto = _doctorRepository.CreateDoctor(doctorDto);
+            return Ok(createdDoctorDto);
```

Comment:
there is a missed requirement here 
![image](https://github.com/RanaSalameh20/DoctorWho.WebAPI/assets/35167530/fd6c5688-5d77-49be-85f3-f560ea3d21fc)